lib/ts/recipe/multifactorauth/multiFactorAuthClaim.ts

```diff
@@ -0,0 +1,257 @@
+        //   2. if requirementList is ["f1", { allOfInAnyOrder: ["f2", "f3"] }, "f4"] and user has completed f1, f2, this functions returns the group ["f2", "f3"]
+        //   3. if requirementList is [ oneOf: ["f1", "f2"], allofInAnyOrder: ["f3", "f4"], "f5" ] and user has completed f1, f3, this functions returns the group ["f3", "f4"] since that's the first group of factors which is not satisfied
+
+        for (const req of requirementList) {
+            const nextFactors: Set<string> = new Set();
+            let type: "string" | "oneOf" | "allOfInAnyOrder" = "string";
+
+            if (typeof req === "string") {
+                if (completedFactors[req] === undefined) {
+                    type = "string";
+                    nextFactors.add(req);
+                }
+            } else if ("oneOf" in req) {
+                let satisfied = false;
+                for (const factorId of req.oneOf) {
+                    if (completedFactors[factorId] !== undefined) {
+                        satisfied = true;
+                    }
+                }
+                if (!satisfied) {
+                    type = "oneOf";
+                    for (const factorId of req.oneOf) {
+                        nextFactors.add(factorId);
+                    }
+                }
+            } else if ("allOfInAnyOrder" in req) {
+                for (const factorId of req.allOfInAnyOrder) {
+                    type = "allOfInAnyOrder";
+                    if (completedFactors[factorId] === undefined) {
+                        nextFactors.add(factorId);
+                    }
+                }
+            }
+            if (nextFactors.size > 0) {
+                return {
+                    factorIds: Array.from(nextFactors),
+                    type: type,
+                };
+            }
+        }
+
+        return {
+            factorIds: [],
+            type: "string",
+        };
+    }
+
+    public fetchValue = async (
+        userId: string,
+        _recipeUserId: RecipeUserId,
+        tenantId: string,
+        currentPayload: JSONObject | undefined,
+        userContext: UserContext
+    ) => {
+        const mfaInfo = await getMFARelatedInfoFromSession({
+            userId,
+            tenantId,
+            accessTokenPayload: currentPayload,
+            assumeEmptyCompletedIfNotFound: true,
+            userContext,
+        });
+
+        if (mfaInfo.status === "OK") {
+            let { completedFactors, mfaRequirementsForAuth } = mfaInfo;
+            return {
+                c: completedFactors,
+                v: this.getNextSetOfUnsatisfiedFactors(completedFactors, mfaRequirementsForAuth).factorIds.length === 0,
+            };
+        } else if (mfaInfo.status === "MFA_CLAIM_VALUE_NOT_FOUND_ERROR") {
+            throw new Error("should never happen"); // because we assume missing claim value as empty completed factors in the function `getMFARelatedInfoFromSession`
```

Comment:
why would this never happen? If the session is an older one, it can come here right?